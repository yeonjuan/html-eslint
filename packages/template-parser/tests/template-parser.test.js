const templateParser = require("../lib/template-parser");
const espree = require("espree");

const parseCode = (code) =>
  espree.parse(code, {
    range: true,
    loc: true,
    ecmaVersion: "latest",
  });

describe("parseTemplate", () => {
  test("simple", () => {
    const code = "`<div></div>`;";
    const ast = parseCode(code);
    const exp = ast.body[0].expression;

    expect(templateParser.parse(exp)).toMatchSnapshot();
  });

  test("newline", () => {
    const code = `\`<div>
</div>\`;`;
    const ast = parseCode(code);
    const exp = ast.body[0].expression;
    expect(templateParser.parse(exp)).toMatchSnapshot();
  });

  test("padding", () => {
    const code = `const html = \`<div>
</div>\`;`;
    const ast = parseCode(code);
    const exp = ast.body[0].declarations[0].init;
    expect(templateParser.parse(exp)).toMatchSnapshot();
  });
});
