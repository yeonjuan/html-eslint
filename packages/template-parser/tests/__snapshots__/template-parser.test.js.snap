// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseTemplate newline 1`] = `
{
  "children": [
    {
      "attributes": [],
      "children": [
        {
          "loc": {
            "end": {
              "column": 8,
              "line": 2,
            },
            "start": {
              "column": 13,
              "line": 1,
            },
          },
          "range": [
            6,
            7,
          ],
          "type": "Text",
          "value": "
",
        },
      ],
      "close": {
        "loc": {
          "end": {
            "column": 6,
            "line": 2,
          },
          "start": {
            "column": 0,
            "line": 2,
          },
        },
        "range": [
          7,
          13,
        ],
        "type": "CloseTag",
        "value": "</div>",
      },
      "loc": {
        "end": {
          "column": 6,
          "line": 2,
        },
        "start": {
          "column": 8,
          "line": 1,
        },
      },
      "name": "div",
      "openEnd": {
        "loc": {
          "end": {
            "column": 13,
            "line": 1,
          },
          "start": {
            "column": 12,
            "line": 1,
          },
        },
        "range": [
          5,
          6,
        ],
        "type": "OpenTagEnd",
        "value": ">",
      },
      "openStart": {
        "loc": {
          "end": {
            "column": 12,
            "line": 1,
          },
          "start": {
            "column": 8,
            "line": 1,
          },
        },
        "range": [
          1,
          5,
        ],
        "type": "OpenTagStart",
        "value": "<div",
      },
      "range": [
        1,
        13,
      ],
      "selfClosing": false,
      "type": "Tag",
    },
  ],
  "loc": {
    "end": {
      "column": 6,
      "line": 2,
    },
    "start": {
      "column": 8,
      "line": 1,
    },
  },
  "range": [
    0,
    13,
  ],
  "type": "Document",
}
`;

exports[`parseTemplate padding 1`] = `
{
  "children": [
    {
      "attributes": [],
      "children": [
        {
          "loc": {
            "end": {
              "column": 8,
              "line": 2,
            },
            "start": {
              "column": 13,
              "line": 1,
            },
          },
          "range": [
            19,
            20,
          ],
          "type": "Text",
          "value": "
",
        },
      ],
      "close": {
        "loc": {
          "end": {
            "column": 6,
            "line": 2,
          },
          "start": {
            "column": 0,
            "line": 2,
          },
        },
        "range": [
          20,
          26,
        ],
        "type": "CloseTag",
        "value": "</div>",
      },
      "loc": {
        "end": {
          "column": 6,
          "line": 2,
        },
        "start": {
          "column": 8,
          "line": 1,
        },
      },
      "name": "div",
      "openEnd": {
        "loc": {
          "end": {
            "column": 13,
            "line": 1,
          },
          "start": {
            "column": 12,
            "line": 1,
          },
        },
        "range": [
          18,
          19,
        ],
        "type": "OpenTagEnd",
        "value": ">",
      },
      "openStart": {
        "loc": {
          "end": {
            "column": 12,
            "line": 1,
          },
          "start": {
            "column": 8,
            "line": 1,
          },
        },
        "range": [
          14,
          18,
        ],
        "type": "OpenTagStart",
        "value": "<div",
      },
      "range": [
        14,
        26,
      ],
      "selfClosing": false,
      "type": "Tag",
    },
  ],
  "loc": {
    "end": {
      "column": 6,
      "line": 2,
    },
    "start": {
      "column": 8,
      "line": 1,
    },
  },
  "range": [
    0,
    26,
  ],
  "type": "Document",
}
`;

exports[`parseTemplate simple 1`] = `
{
  "children": [
    {
      "attributes": [],
      "children": [],
      "close": {
        "loc": {
          "end": {
            "column": 25,
            "line": 1,
          },
          "start": {
            "column": 19,
            "line": 1,
          },
        },
        "range": [
          6,
          12,
        ],
        "type": "CloseTag",
        "value": "</div>",
      },
      "loc": {
        "end": {
          "column": 25,
          "line": 1,
        },
        "start": {
          "column": 14,
          "line": 1,
        },
      },
      "name": "div",
      "openEnd": {
        "loc": {
          "end": {
            "column": 19,
            "line": 1,
          },
          "start": {
            "column": 18,
            "line": 1,
          },
        },
        "range": [
          5,
          6,
        ],
        "type": "OpenTagEnd",
        "value": ">",
      },
      "openStart": {
        "loc": {
          "end": {
            "column": 18,
            "line": 1,
          },
          "start": {
            "column": 14,
            "line": 1,
          },
        },
        "range": [
          1,
          5,
        ],
        "type": "OpenTagStart",
        "value": "<div",
      },
      "range": [
        1,
        12,
      ],
      "selfClosing": false,
      "type": "Tag",
    },
  ],
  "loc": {
    "end": {
      "column": 25,
      "line": 1,
    },
    "start": {
      "column": 14,
      "line": 1,
    },
  },
  "range": [
    0,
    12,
  ],
  "type": "Document",
}
`;
